nest-tax-backend/src/utils/logging.ts

```diff
@@ -0,0 +1,189 @@
+ * a key will be replaced by their descriptions.
+ */
+export function separateLogFromResponseObj(obj: object) {
+  const responseLog: Record<string, unknown> = {}
+  const responseMessage: Record<string, unknown> = {}
+
+  const keys = Object.getOwnPropertyNames(obj)
+  const symbols = Object.getOwnPropertySymbols(obj)
+
+  keys.forEach((key) => {
+    if (key.startsWith('$Symbol-')) {
+      responseLog[key.slice(8)] = (obj as any)[key]
+    } else {
+      responseMessage[key] = (obj as any)[key]
+    }
+  })
+
+  symbols.forEach((symbol) => {
+    if (symbol.description) {
+      responseLog[symbol.description] = (obj as any)[symbol]
+    }
+  })
+
+  return { responseLog, responseMessage }
+}
+
+/**
+ * Takes and object and creates a flat one-line string representation.
+ *
+ * Example output
+ *  "key1"="value1" "key2"="value2" "key3"="{ objectKey1: \"string with newline at the end\n\"}"
+ *
+ * If an object is present as a value, it will be converted to json string.
+ * Return string does not contain any new line symbols - they will be replaced by '\n'.
+ *
+ * If a key is represented as a symbol, it will be included in the final output.
+ */
+export function objToLogfmt(obj: object): string {
+  const separatedValues = separateLogFromResponseObj(obj)
+  const objAll = {
+    ...separatedValues.responseLog,
+    ...separatedValues.responseMessage,
+  }
+  return Object.entries(objAll)
+    .map(([key, value]) => {
+      let newValue = value
+      if (typeof newValue === 'object') {
+        newValue = JSON.stringify(newValue)
+      }
+      // not else if because we want to sanitize strings in object
+      if (typeof newValue === 'string') {
+        newValue = escapeForLogfmt(newValue)
+      }
+
+      return `${key}="${newValue}"`
+    })
+    .join(' ')
+}
+
+function httpExceptionToObj(
+  error: HttpException,
+  methodName?: string | symbol,
+): object {
+  const response = error.getResponse()
+  try {
+    const { responseLog, responseMessage } = separateLogFromResponseObj(
+      typeof response === 'string' ? JSON.parse(response) : response,
+    )
+    return {
+      errorType: error.name,
+      ...responseMessage,
+      ...responseLog,
+      method: methodName,
+      stack: error.stack,
+    }
+  } catch (parseError) {
+    return {
+      errorType: error.name,
+      message: error.message,
+      method: methodName,
+      stack: error.stack,
+    }
+  }
+}
+
+function requiredErrorToObj(
+  error: RequiredError,
+  methodName?: string | symbol,
+): object {
+  return {
+    errorType: 'RequiredError',
+    message: error.message,
+    field: error.field,
+    method: methodName,
+    stack: error.stack,
+  }
+}
+
+function genericErrorToObj(error: Error, methodName?: string | symbol): object {
+  return {
+    errorType: error.name,
+    message: error.message,
+    method: methodName,
+    stack: error.stack,
+  }
+}
+
+export function errorToLogfmt(
```

Comment:
This is a good candidate how to avoid let:
```
function createLogObj(
  error: unknown,
  methodName?: string | symbol
): object {
  if (error instanceof HttpException) {
    return httpExceptionToObj(error, methodName);
  } else if (error instanceof RequiredError) {
    return requiredErrorToObj(error, methodName);
  } else if (error instanceof Error) {
    return genericErrorToObj(error, methodName);
  } else {
    return {
      errorType: 'UnknownError',
      message: 'Unknown error was thrown. This should not happen',
      method: methodName,
      alert: 1,
    };
  }
}

export function errorToLogfmt(
  error: unknown,
  methodName?: string | symbol,
): string {
  return objToLogfmt(createLogObj(error, methodName));
}
```

Don't be afraid to create more functions and then compose them.